### PR TITLE
Use EnableCodeStyleSeverity instead of AnalysisLevel to control new diagnostic severity behavior

### PR DESCRIPTION
**Pull request by sharwell**
* Use built-in features for marking unnecessary code
* Verify missing code style severity falls back to old behavior
* Use `EnableCodeStyleSeverity` instead of `AnalysisLevel` to control new diagnostic severity behavior

Follow-up to #72238

### CODE REVIEW

**Review by arkalyanms — APPROVED** [review]

**Comment by cremor** [comment]
If I understand this PR correctly then these documentation pages need to be updated for the change, correct?
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/overview#enable-on-build
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/language-rules#option-format

**Comment by sharwell** [comment]
Yes, those would need to be updated. We'll also need to update the SDK code style build targets.

File: src/Analyzers/CSharp/Tests/UseObjectInitializer/UseObjectInitializerTests.cs [hunk 1]
```diff
@@ -20,17 +20,6 @@ namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.UseObjectInitializer;
 [Trait(Traits.Feature, Traits.Features.CodeActionsUseObjectInitializer)]
 public partial class UseObjectInitializerTests
 {
-    private static async Task TestInRegularAndScriptAsync(string testCode, string fixedCode, OutputKind outputKind = OutputKind.DynamicallyLinkedLibrary)
```
Comment (jaredpar) [inline src/Analyzers/CSharp/Tests/UseObjectInitializer/UseObjectInitializerTests.cs:23]: Why did this change produce so much churn in the object init tests?

Comment (sharwell) [reply src/Analyzers/CSharp/Tests/UseObjectInitializer/UseObjectInitializerTests.cs:23]: This analyzer was using an old strategy for code fading that was interfering with some new testing we needed for the feature change. I converted the analyzer to the new style as a separate commit at the beginning of this PR.

**Review by chsienki — APPROVED** [review]

**Review by 333fred — APPROVED** [review]

**Review by jjonescz — APPROVED** [review]

**Comment by sharwell** [comment]
Rebased after reviews completed to apply the fixup commit.

**Pull request merged**